client/state/selectors/get-removable-connections.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+
+import { filter } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentUserId } from 'state/current-user/selectors';
+import { getKeyringConnectionsByName } from 'state/sharing/keyring/selectors';
+import { getRemovableConnections as getRemovablePublicizeConnections } from 'state/sharing/publicize/selectors';
+
+/**
+ * Given a Keyring service name, returns the connections that the current user is
+ * allowed to remove.
+ *
+ * For them to be allowed to remove a connection they need to have either the
+ * `edit_others_posts` capability or it's a connection to one of
+ * their accounts.
+ *
+ * @param   {object} state   Global state tree
+ * @param   {string} service The name of the service
+ * @returns {Array}          Connections that the current user is allowed to remove
+ */
+export default function getRemovableConnections( state, service ) {
```

Comment:
Do you think we should add a test for this?

Reply:
Reason why I didn't is because this fundamentally merges two selectors, each with their own tests.
I guess adding more tests for this doesn't do any harm though, and it should also be convenient, by copying from the other tests!